### ISSUE TITLE
Final review

### ISSUE THREAD

**Issue by MasonGallo** [marymitchellnj/MM_Data_Science_Portfolio#4]
Clarity of research problem:

I think your slides are very clear. I did not see any conclusions at time of review.

Code:

Your code is clear with tons of comments. In your question about NB, remember that it doesn't matter what numbers you use to encode the response variable as long as they're consistent. NB doesn't require a binary response variable. You can use 1,2,3.

Next steps:

Your next step should be to setup a benchmark to measure your performance and evaluate several models. Don't forget to look up solutions to the kaggle problem.

**Comment by marymitchellnj** [comment]
Thank you, Mason, for your helpful comments. I shared them with Lema today when we met during office hours. Your last statement was an incredible tip. I had been grappling with some quality issues (especially words smashed together here and there) in the product_descriptions field that were not evident initially (only after I had pulled a 1000-row sample into Excel to read in detail the columns that were truncated in the Python Pandas dataframe. I first double-checked my own process to make sure I had not caused that. We then checked the area of the Kaggle site for this project and found that other Kagglers had raised this as a data quality issue on Home Depot's part with the product_descriptions column. One of them had suggested a workaround, but in R. I'm now trying to convert that command from R to Python. It is (in R): 

gsub("([a-z])([A-Z])([a-z])", "\\1 \\2\\3", product_description)

I think I need to use df.replace instead of the "gsub" so will try that. 

Meanwhile, I'm moving ahead with the steps you suggested. Thanks for clarifying re: Naive Bayes accepting more than two options. 

Mary

**Comment by MasonGallo** [comment]
Yup, df.replace will accept regex arguments so that should work for your purposes.
